### PR TITLE
feat: add ability to specify offsets for JSHandle.click

### PR DESCRIPTION
**Pull request by OrKoN**
Until now, the click would be always sent to the middle
point of the target element. With this change, one can define
offsets relative to the border box of the elements and click
different areas of an element.

### CODE REVIEW

**Comment by OrKoN** [comment]
I am still looking into the doclint failures but the code is ready to be reviewed I think. If you can tell what's wrong with the docs, I'd really appreciate :)

**Review by jschfflr — APPROVED** [review]
Looks good to me!

File: docs/api.md
Comment (jschfflr) [inline docs/api.md]: I think by overwriting this name in `check_public_api`, it now expects this to be of type `Object`

**Review by jschfflr — APPROVED** [review]

File: docs/api.md
Comment (OrKoN) [reply docs/api.md]: it seems to complain nevertheless. Let's check on CI.

File: utils/doclint/check_public_api/index.js
Comment (jschfflr) [inline utils/doclint/check_public_api/index.js]:
```suggestion
        'Method ElementHandle.clickablePoint() offset',
```

Comment (jschfflr) [reply utils/doclint/check_public_api/index.js]: I think this should resolve the doc lint error.

Comment (OrKoN) [reply utils/doclint/check_public_api/index.js]: wow, thanks!

**Comment by jschfflr** [comment]
Should we get this landed or wait for a review from @mathiasbynens?

File: docs/api.md
Comment (mathiasbynens) [inline docs/api.md]: Relative to which corner (top left or bottom left)? Please clarify this in the docs.

**Review by mathiasbynens — APPROVED** [review]
LGTM % doc clarification

**Pull request merged**
